packages/desktop-client/src/components/payees/PayeeTable.tsx

```diff
@@ -1,31 +1,50 @@
-import { forwardRef, useState, useLayoutEffect, useCallback } from 'react';
+import {
+  forwardRef,
+  useCallback,
+  useLayoutEffect,
+  useState,
+  type ComponentProps,
+} from 'react';
+
+import { type PayeeEntity } from 'loot-core/src/types/models';
 
 import { useSelectedItems } from '../../hooks/useSelected';
 import View from '../common/View';
-import { Table } from '../table';
+import { Table, type TableNavigator } from '../table';
 
 import PayeeTableRow from './PayeeTableRow';
 
+// Table items require an ID to work, it's optional in the loot-core
+// model so would need to verify accuracy of that before changing there
+type PayeeWithId = PayeeEntity & Required<Pick<PayeeEntity, 'id'>>;
+
+type PayeeTableProps = {
+  payees: PayeeWithId[];
+  ruleCounts: Map<PayeeWithId['id'], number>;
+  navigator: TableNavigator<PayeeWithId>;
+} & Pick<
+  ComponentProps<typeof PayeeTableRow>,
+  'onUpdate' | 'onViewRules' | 'onCreateRule'
+>;
+
 const PayeeTable = forwardRef(
   (
     {
       payees,
       ruleCounts,
       navigator,
-      categoryGroups,
-      highlightedRows,
-      ruleActions,
       onUpdate,
       onViewRules,
       onCreateRule,
-    },
-    ref,
+    }: PayeeTableProps,
+    ref: ComponentProps<typeof Table>['ref'],
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏would be better to set it as a param to `forwardRef`.

Example: https://github.com/actualbudget/actual/blob/baa474843a20a26c8a76317ea1b506d814d71f78/packages/desktop-client/src/components/table.tsx#L68

Reply:
That was my original plan too, but I couldn't figure out how to pull just the inner type of the `Ref` up from `Table` which is why I moved it down to the parameter here.

I could instead export the `TableHandleRef` from `table.tsx` and use it directly, but that does lose the nice automatic type inheritance.

Reply:
You can use `React.ComponentRef<typeof Table>`

Reply:
Ah perfect thank you, that's another one for my tool belt.

Fixing the type on this has quickly led me back to an underlying issue with the `Table` typing: the ref type and the props type should both be passed the generic `T` - which doesn't work out of the box with `forwardRef` as it's not a plain function type.

This is a good summary of the root problem with different solutions: https://fettblog.eu/typescript-react-generic-forward-refs/. Do you have any preference which approach to take here? I personally most like option 3, and least like option 1. May as well fix this while I'm touching the code.

Reply:
In face, for now I've just pushed this using the same trick the props is (setting the default generic type to be the `TableItem`).

Can follow this up in an issue.